common/persistence/domainReplicationQueue.go

```diff
@@ -110,18 +116,63 @@ func (q *domainReplicationQueueImpl) UpdateAckLevel(lastProcessedMessageID int,
 		return fmt.Errorf("failed to update ack level: %v", err)
 	}
 
-	select {
-	case q.ackNotificationChan <- true:
-	default:
-	}
-
 	return nil
 }
 
 func (q *domainReplicationQueueImpl) GetAckLevels() (map[string]int, error) {
 	return q.queue.GetAckLevels()
 }
 
-func (q *domainReplicationQueueImpl) Close() {
-	close(q.done)
+func (q *domainReplicationQueueImpl) purgeAckedMessages() error {
+	ackLevelByCluster, err := q.GetAckLevels()
+	if err != nil {
+		return fmt.Errorf("failed to purge messages: %v", err)
+	}
+
+	if len(ackLevelByCluster) == 0 {
+		return nil
+	}
+
+	minAckLevel := math.MaxInt64
+	for _, ackLevel := range ackLevelByCluster {
+		if ackLevel < minAckLevel {
+			minAckLevel = ackLevel
+		}
+	}
+
+	err = q.queue.DeleteMessagesBefore(minAckLevel)
+	if err != nil {
+		return fmt.Errorf("failed to purge messages: %v", err)
+	}
+
+	q.metricsClient.
+		Scope(metrics.FrontendDomainReplicationQueueScope).
```

Comment:
Using frontendScope within the common/persistence layer seems weird. Is this a copy/paste error ?

Reply:
Human error. Update the name of the scope

Reply:
The only trigger from frontend. Keep the frontendScope for now